packages/graphql/src/schema/resolvers/subscriptions/where/filters/filter-by-authorization-rules.ts

```diff
@@ -0,0 +1,148 @@
+                    node,
+                    where: wherePropertyValue as Record<string, RecordType>,
+                    event,
+                    nodes,
+                    relationshipFields,
+                    context,
+                });
+            } else {
+                comparisonResults = (wherePropertyValue as Array<Record<string, RecordType>>).map((whereCl) => {
+                    return filterByAuthorizationRules({
+                        node,
+                        where: whereCl,
+                        event,
+                        nodes,
+                        relationshipFields,
+                        context,
+                    });
+                });
+            }
+
+            if (!comparisonResultsAggregationFn(comparisonResults)) {
+                return false;
+            }
+        }
+
+        if (wherePropertyKey === "node") {
+            switch (event.event) {
+                case "create":
+                    return filterByProperties({
+                        node,
+                        whereProperties: wherePropertyValue,
+                        receivedProperties: event.properties.new,
+                    });
+                case "update":
+                case "delete":
+                    return filterByProperties({
+                        node,
+                        whereProperties: wherePropertyValue,
+                        receivedProperties: event.properties.old,
+                    });
+                case "create_relationship":
+                case "delete_relationship": {
+                    const receivedEventRelationshipType = event.relationshipName;
+                    const relationships = node.relationFields.filter((f) => f.type === receivedEventRelationshipType);
+                    if (!relationships.length) {
+                        return false;
+                    }
+                    const receivedEventRelationship = relationships[0] as RelationField; // ONE relationship only possible
+                    const key = receivedEventRelationship.direction === "IN" ? "to" : "from";
+                    return filterByProperties({
+                        node,
+                        whereProperties: wherePropertyValue,
+                        receivedProperties: receivedEventProperties[key],
+                    });
+                }
+            }
+        }
+
+        if (wherePropertyKey === "relationship") {
+            if (!nodes || !relationshipFields || !isRelationshipSubscriptionsEvent(event)) {
+                return false;
+            }
+
+            const receivedEventRelationshipType = event.relationshipName;
+            const relationships = node.relationFields.filter((f) => f.type === receivedEventRelationshipType);
+            if (!relationships.length) {
+                return false;
+            }
+            const receivedEventRelationship = relationships[0] as RelationField; // ONE relationship only possible
```

Comment:
```suggestion
    const receivedEventRelationship = relationships[0]; // ONE relationship only possible
    if (!receivedEventRelationship) {
        return false;
    }
```

With this we can allow Typescript to narrow the type to `RelationField`

Reply:
Nice!